x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/enrollment/CreateEnrollmentTokenRequest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.enrollment;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Represents a request to create an enrollment token based on the provided information.
+ */
+
+public class CreateEnrollmentTokenRequest extends ActionRequest {
+    private final String id;
+
+    public CreateEnrollmentTokenRequest(StreamInput in) throws IOException {
+        super(in);
+        this.id = UUIDs.base64UUID();
```

Comment:
Why do we need an id field and what does it do ?

Reply:
Leftover from initial testing

Reply:
Yeap, figured this out as I read the rest of the PR but failed to come back and remove this comment. I still think we can use a CreateApiRequest internally so the the CreateEnrollmentTokenRequest would not need an id